operator/src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -95,101 +97,103 @@ public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final
-                                        )
-                                        .endMetadata()
-                                        .withData(data)
-                                        .build()
-                        )
-                        .create()
-            );
+                        .withName(secretName)
+                        .edit(s -> new SecretBuilder(s).withData(data).build());
+            }
+        } else {
+            LOGGER.info("Creating new secret {}/{}", kafkaAccessNamespace, secretName);
+            kubernetesClient
+                    .secrets()
+                    .inNamespace(kafkaAccessNamespace)
+                    .resource(
+                            new SecretBuilder()
+                                    .withType(SECRET_TYPE)
+                                    .withNewMetadata()
+                                    .withName(secretName)
+                                    .withLabels(commonSecretLabels)
+                                    .withOwnerReferences(
+                                            new OwnerReferenceBuilder()
+                                                    .withApiVersion(kafkaAccess.getApiVersion())
+                                                    .withKind(kafkaAccess.getKind())
+                                                    .withName(kafkaAccessName)
+                                                    .withUid(kafkaAccess.getMetadata().getUid())
+                                                    .withBlockOwnerDeletion(false)
+                                                    .withController(false)
+                                                    .build()
+                                    )
+                                    .endMetadata()
+                                    .withData(data)
+                                    .build()
+                    )
+                    .create();
+        }
     }
 
     /**
-     * Prepares the event sources required for triggering the reconciliation
+     * Prepares the event sources required for triggering the reconciliation. This tells the JOSDK framework which resources the operator needs to watch.
      *
      * @param context       The EventSourceContext for KafkaAccess resource
      *
      * @return              A new map of event sources
      */
-    @Override
-    public Map<String, EventSource> prepareEventSources(final EventSourceContext<KafkaAccess> context) {
+    public List<EventSource<?, KafkaAccess>> prepareEventSources(EventSourceContext<KafkaAccess> context) {
         LOGGER.info("Preparing event sources");
-        InformerEventSource<Kafka, KafkaAccess> kafkaEventSource = new InformerEventSource<>(
-                InformerConfiguration.from(Kafka.class, context)
+        InformerEventSourceConfiguration<Kafka> kafkaEventSource =
+                // now using updated in v5: InformerEventSourceConfiguration
```

Comment:
This would be a good comment made via the PR review window, but not needed in the actual code going forwards
```suggestion
```